### PR TITLE
fix(query builder): Throw error on missing or unknown properties

### PR DESCRIPTION
**Pull request by vlapo**
- handle emptiness and wrong properties in update query builder
 - fix https://github.com/typeorm/typeorm/issues/2849

### CODE REVIEW

File: src/error/UpdateValuesMissingError.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Thrown when user tries to update using QueryBuilder but do not specify what to update.
+ */
+export class UpdateValuesMissingError extends Error {
+    name = "InsertValuesMissingError";
```
Comment (pleerock) [inline src/error/UpdateValuesMissingError.ts]: name is wrong

File: src/error/UpdateValuesMissingError.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+    constructor() {
+        super();
+        Object.setPrototypeOf(this, UpdateValuesMissingError.prototype);
+        this.message = `Cannot perform update query because update values are not defined. Call "qb.set(...)" method to specify inserted values.`;
```
Comment (pleerock) [inline src/error/UpdateValuesMissingError.ts]: "updated values"

File: src/error/ColumnMetadataNotFound.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/**
+ *
+ */
+export class ColumnMetadataNotFoundError extends Error {
+    name = "ColumnMetadataNotFoundError";
+
+    constructor(propertyPath: string) {
+        super();
+        Object.setPrototypeOf(this, ColumnMetadataNotFoundError.prototype);
+        this.message = `No column metadata for property "${propertyPath}" was found.`;
```
Comment (pleerock) [inline src/error/ColumnMetadataNotFound.ts]: probably we need to give more user-friendly error. Users don't know what "metadata" means.

Comment (vlapo) [reply src/error/ColumnMetadataNotFound.ts]: You are right. We need more user friendly errors. What do you suggest?
EntityColumnNotFound -> `No entity column "${propertyPath}" was found.`?

**Comment by vlapo** [comment]
Totally forgot. Updated by comments.

**Comment by pleerock** [comment]
Thank you. This is important to mention in changelog. In can break some exist code in a way.

**Pull request merged**